### PR TITLE
bugfix: key-auth plugin cannot run accurately in the case of multiple consumers(branch: master)

### PR DESCRIPTION
**Pull request by Beckjiang**
### Summary
fix the logical error in `apisix/consumer.lua` and improve the test case.

### Full changelog

* updated apisix/consumer.lua:40-46
* updated the test case [TEST 7: valid consumer] - t/plugin/key-auth.t 

### Issues resolved
The improved test case run in branch master.
image

Run in this branch.
image

### CODE REVIEW

File: t/plugin/key-auth.t [hunk 1]
```diff
@@ -172,11 +172,34 @@ GET /hello
...
+            ngx.say(body)
+        }
+    }
 --- request
-GET /hello
+GET /add_more_consumer
+--- pipelined_requests eval
+["GET /add_more_consumer", "GET /hello"]
 --- more_headers
 apikey: auth-one
```
Comment (membphis) [inline t/plugin/key-auth.t]: `apikey: auth-one` -> `apikey: auth-13`, test the new code logic.

**Comment by membphis** [comment]
@Beckjiang And please update the version to `0.3`.

https://github.com/apache/incubator-apisix/pull/826/files#diff-d9f05f02182918b9c75a07044308b731R26

File: t/plugin/key-auth.t (hunk 1 above)
Comment (Beckjiang) [reply t/plugin/key-auth.t]: image

YES,UPDATED

**Comment by membphis** [comment]
@Beckjiang Merged, many thanks.

**Pull request merged**